### PR TITLE
Make the default date for reminders the 15th

### PR DESCRIPTION
**Pull request by TomPretty**
## Why are you doing this?

Make the default date for post contribution reminders the 15th of the month, as per [this card](https://trello.com/c/ql304o5j/2361-update-the-post-contribution-reminder-date-to-be-the-15th-of-each-month-from-nov-instead-of-the-1st-to-be-able-to-send-the-email)

**FOLLOW UP**
- manually update reminders after from November onwards to be on the 15th (in braze + contribution_reminders) (BUT NOT THE ONES FOR GIVING TUESDAY)

**NB**: I uncovered a bug whilst implementing this that means all reminders set for the last day of the year were actually set to the 1st of December

### CODE REVIEW

File: support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx [hunk 1]
```diff
@@ -140,7 +142,9 @@ const ContributionThankYouSupportReminder = ({
     const year = selectedDate.getFullYear();
     const month = selectedDate.getMonth() + 1; // javascript dates run from 0-11, we want 1-12
     const paddedMonth = month.toString().padStart(2, '0');
-    return `${year}-${paddedMonth}-01 00:00:00`;
```
Comment (TomPretty) [inline support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx:143]: This has lead to a bug with the reminders for the us eoy appeal. Reminders set for the last day of the year, actually got set to the 1st of December (same as Giving Tuesday)

**Review by paulbrown1982 — APPROVED** [review]
👍  Let's get this merged!

File: support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx [hunk 2]
```diff
@@ -94,12 +96,12 @@ const BOTH: ReminderDate = {
...
 
 const IN_THREE_MONTHS: ReminderDate = {
-  date: new Date(2021, 2),
+  date: new Date(2021, 2, DAY_OF_MONTH_OF_DEFAULT_REMINDER),
   label: 'in three months (March 2021)', // slightly custom copy over default (three vs 3)
 };
 
 const NEXT_US_EOY_APPEAL: ReminderDate = {
-  date: new Date(2021, 11),
+  date: new Date(2021, 11, DAY_OF_MONTH_OF_DEFAULT_REMINDER),
```
Comment (paulbrown1982) [inline support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx:104]: Is `DAY_OF_MONTH_OF_DEFAULT_REMINDER ` right here?

Comment (TomPretty) [reply support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx:104]: Oh actually, yeah good point - this should probably just be on the 1st

Comment (paulbrown1982) [reply support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx]: Shouldn't it be 31st?

Comment (TomPretty) [reply support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx]: yeah it should be!

**Review by paulbrown1982 — APPROVED** [review]
👍  LGTM now

**Pull request merged**